docs/template_plugin/src/template_infer_request.hpp

```diff
@@ -46,6 +46,13 @@ class TemplateInferRequest : public InferenceEngine::InferRequestInternal {
     void waitPipeline();
     void inferPostprocess();
 
+    void GetBlob(const char* name, InferenceEngine::Blob::Ptr& data) override;
+
+    void SetShape(const char* name, const InferenceEngine::SizeVector& dims) override {
+        // TODO: check partial shape compatibility
+        m_realShapes[name] = dims;
```

Comment:
should not we check that this is valid only for dynamic shapes? E.g. if we have usual input 20x20, but set shape in called with 10x10, an exception should thrown?